### PR TITLE
Implement Process Exits Block Operation

### PR DESCRIPTION
**Pull request by rauljordan**
Part of #1073 

---

# Description

This PR implements a ProcessValidatorExits operation as specified by the Ethereum 2.0 specification in the block operations section [here](https://github.com/ethereum/eth2.0-specs/tree/e2714b3344af95fe3c6fe42da6aac61c02172ee4/specs/core/0_beacon-chain.md#exits-1) as of commit [e2714b3344af95fe3c6fe42da6aac61c02172ee4](https://github.com/ethereum/eth2.0-specs/tree/e2714b3344af95fe3c6fe42da6aac61c02172ee4/specs/core/0_beacon-chain.md#exits-1) in the master branch of the spec.

### CODE REVIEW

File: beacon-chain/core/blocks/block_operations.go [hunk 1]
```diff
@@ -264,3 +266,94 @@ func verifyCasperVotes(votes *pb.SlashableVoteData) error {
...
 }
+
+// ProcessValidatorExits is one of the operations performed
+// on each processed beacon block to determine which validators
+// should exit the state's validator registry.
+//
+// Official spec definition for processing exits:
+//
+//   Verify that len(block.body.casper_slashings) <= MAX_CASPER_SLASHINGS.
+//   For each casper_slashing in block.body.casper_slashings:
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/block_operations.go]: what does this have to do with exits?

Comment (prestonvanloon) [reply beacon-chain/core/blocks/block_operations.go]: I think this is a copy paste error, no?

File: beacon-chain/core/blocks/block_operations.go [hunk 2]
```diff
@@ -428,3 +429,91 @@ func verifyAttestation(beaconState *types.BeaconState, att *pb.Attestation) erro
...
+	if beaconState.Slot() < exit.GetSlot() {
+		return fmt.Errorf(
+			"expected state.Slot >= exit.Slot, received %d < %d",
+			beaconState.Slot(),
+			exit.GetSlot(),
+		)
+	}
+	persistentCommitteeSlot := validator.GetLatestStatusChangeSlot() +
+		params.BeaconConfig().ShardPersistentCommitteeChangePeriod
+	fmt.Println(persistentCommitteeSlot)
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/block_operations.go]: Remove

Comment (rauljordan) [reply beacon-chain/core/blocks/block_operations.go]: kk ready again @prestonvanloon
